src/Paprika/Store/StorageFanOutPage.cs

```diff
@@ -102,6 +102,32 @@ public void Report(IReporter reporter, IPageResolver resolver, int level)
             }
         }
     }
+    public Page Destroy(IBatchContext batch, in NibblePath prefix)
+    {
+        if (Header.BatchId != batch.BatchId)
+        {
+            // the page is from another batch, meaning, it's readonly. Copy
+            var writable = batch.GetWritableCopy(page);
+            return new StorageFanOutPage<TNext>(writable).Set(prefix, ReadOnlySpan<byte>.Empty, batch);
```

Comment:
Each method that modified things, needs to COW. To Copy-On-Write, you do the writable thing and then you call the same method again, so that it should be `Destroy` called here.

Reply:
Have implemented it like this - 

`if (Header.BatchId != batch.BatchId)
        {
            // the page is from another batch, meaning, it's readonly. Copy
            var writable = batch.GetWritableCopy(page);
            return new StorageFanOutPage<TNext>(writable).Destroy(batch, prefix);
        }
`
Looks good?